actors.md

```diff
@@ -679,68 +668,44 @@ func SubmitPost(proofs PoStProof, faults []FaultSet, recovered Bitfield, done Bi
 		TransferFunds(msg.From, msg.Value-feesRequired)
 	}
 
-	if !CheckPostProof(miner.SectorSize, proof, faults) {
+    var seed
+    if chain.Now() < self.ProvingPeriodEnd {
+      // good case, submitted in time
+      seed = GetRandFromBlock(self.ProvingPeriodEnd - POST_CHALLENGE_TIME)
+    } else {
+      // bad case, submitted late, need to take new proving period end as reference
+      seed = GetRandFromBlock(nextPovingPeriodEnd - POST_CHALLENGE_TIME)
+    }
```

Comment:
There are some obvious cases where this could fail if `SubmitPoSt` is submitted too early, or races against the proving period end arriving. I guess they're implicitly fatal, but the spec should eventually be specific.